resim/metrics/python/metrics_writer_test.py

```diff
@@ -685,6 +686,68 @@ def test_event(self) -> None:
         for metric_id in event_msg.metrics:
             self.assertIn(metric_id, metric_id_uuids)
 
+    def test_override_status(self) -> None:
+        ALL_STATUSES = [
+            MetricStatus.PASSED_METRIC_STATUS,
+            MetricStatus.FAIL_WARN_METRIC_STATUS,
+            MetricStatus.FAIL_BLOCK_METRIC_STATUS,
+            MetricStatus.NO_METRIC_STATUS,
+            MetricStatus.NOT_APPLICABLE_METRIC_STATUS,
+            MetricStatus.RAW_METRIC_STATUS,
+        ]
+        for metric_status, override_status in itertools.product(
+            ALL_STATUSES, ALL_STATUSES
+        ):
+            if override_status != metric_status:
+                print(metric_status, override_status)
+                self.writer = ResimMetricsWriter(self.job_id)
+                METRIC_NAME = (
+                    "Scalar metric" + str(metric_status) + str(override_status)
+                )
+                METRIC_VALUE = 5.0
+                (
+                    self.writer.add_scalar_metric(METRIC_NAME)
+                    .with_value(METRIC_VALUE)
+                    .with_status(metric_status)
+                )
+                output = self.writer.write(metrics_status_override=override_status)
+                self.assertEqual(
+                    output.metrics_msg.metrics_status, override_status.value
+                )
+                self.assertEqual(len(output.packed_ids), 1)
+                self.assertEqual(len(output.metrics_msg.job_level_metrics.metrics), 1)
+                self.assertEqual(len(output.metrics_msg.metrics_data), 0)
+
+                metric_base = output.metrics_msg.job_level_metrics.metrics[0]
+                self.assertEqual(metric_base.status, metric_status.value)
+
+                self.writer = ResimMetricsWriter(self.job_id)
+                (
+                    self.writer.add_scalar_metric(METRIC_NAME)
+                    .with_value(METRIC_VALUE)
+                    .with_status(metric_status)
+                )
```

Comment:
I'm pretty sure the `write()` call is non-destructive (although if this were C++ we could be more confident). Can't you therefore re-use the old writer?
```suggestion
```

Reply:
ugh. i'd originally done that, then screwed something and thought this fixed it.